From 68e7f57ada9bca844eb4dffbe6770314de33b96a Mon Sep 17 00:00:00 2001
From: Hatem ElKharashy <hatem.elkharashy@qt.io>
Date: Wed, 9 Aug 2023 12:35:49 +0300
Subject: [PATCH 29/81] Support display=none for Graphical Nodes

According to the documentation, graphical nodes with
display=none should not be drawn. But we only check
the display mode for structure nodes.

Task-number: QTBUG-115946
Change-Id: I211b3f79237967dc95fdadb96a4119893b9e83f4
Reviewed-by: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
Reviewed-by: Matthias Rauter <matthias.rauter@qt.io>
---
 src/svg/qsvgnode.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/svg/qsvgnode.cpp b/src/svg/qsvgnode.cpp
index 31e5338..3914ad6 100644
--- a/src/svg/qsvgnode.cpp
+++ b/src/svg/qsvgnode.cpp
@@ -329,6 +329,10 @@ qreal QSvgNode::strokeWidth(QPainter *p)
 bool QSvgNode::shouldDrawNode(QPainter *p, QSvgExtraStates &states) const
 {
     static bool alwaysDraw = qEnvironmentVariableIntValue("QT_SVG_DISABLE_SIZE_LIMIT");
+
+    if (m_displayMode == DisplayMode::NoneMode)
+        return false;
+
     if (alwaysDraw)
         return true;
 
-- 
2.42.1

